File: 2015/01/sol.py
floor = 0
counter = 1

content = open("input.txt", "r").read()

for char in content:
    if char == "(":
        floor += 1
    elif char == ")":
        floor -= 1

    if floor == -1:
        print(counter)

    counter += 1

print(floor)

File: 2024/3/sol.pl
# Problem statement: https://adventofcode.com/2024/day/3

open my $file, '<', 'input.txt' or die "could not open file: $!";

sub mul {
	my ( $x, $y ) = @_;

	return $x * $y;
}

my $ans = 0;

while ( my $line = <$file> ) {
	chomp($line);

	my @muloperations = $line =~ /mul\(\d+,\d+\)/g;

	foreach my $o (@muloperations) {
		my ( $a, $b ) = $o =~ /(\d+)/g;

		print "$a, $b, $o \t" . mul( $a, $b ) . "\n";

		$ans += mul( $a, $b );
	}
	print "@muloperations\n";
}

print "$ans\n";
